deals-add.js

```diff
@@ -0,0 +1,223 @@
+function calculateSavings(deal) {
+	var savings = parseFloat(deal.listing.previousPrice) - parseFloat(deal.listing.currentPrice);
+	return round(savings, 2); // more accurate rounding method
+}
+
+// add ellipsis where product name cuts off
+function clamp(dealTitle) {
+	$clamp(dealTitle, {clamp: 2});
+}
+
+/* can be used for activities feed in future
+function getTimeAgo(lastCreated) {
+	var date = lastCreated.substring(0, 10);
+	date = date.replace(/-/g, "");
+	date = date + ", " + lastCreated.substring(11, 16);
+
+	return moment(date, "YYYYMMDD, hh:mm").fromNow();
+
+}*/
+
+// change the string to title case
+function toTitleCase(str)
+{
+     return str.replace(/\w\S*/g, 
+    	function(txt){
+    		return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
+    	});
+}
+
+// removes all existing card except the first one, hides the first one
+function resetFeed() {
+	offset = 0;
+	hasMore = true;
+	dealsContainer.firstChild.style.visibility='hidden';
+	var j = 0;
+	for (j = document.getElementsByClassName("deal-card").length; j > 1; j--) {
+		dealsContainer.removeChild(dealsContainer.lastChild);
+	}
+}
+
+// do API call, populate cards with deal info
+function getDeals(filter) {
+	// sets isFetching to true to prevent multiple triggers
+	isFetching = true;
+	$.ajax ({
+		method: 'GET',
+		url: 'https://ratex.co/store/api/products' + '?filter=' + filter
+	})
+	.done(function (response) {
+		// sets address bar with parameters
+		window.history.pushState({urlPath:'/trending-deals-rates?category=' + currentCategory},"",'/trending-deals-rates?category=' + currentCategory);
+		
+		dealsContainer.firstChild.style.visibility='visible';
+		
+		var startOfData = 0;
+		for(i = offset; startOfData < response.data.length && hasMore; i++, startOfData++) {
+			if (i !== startOfData) {
+				createNewCard();
+			}
+			// set product URL
+			document.getElementsByClassName("deal-link")[i].href = response.data[startOfData].listing.merchantURL;
+			
+			// set product image
+			document.getElementsByClassName("deal-img")[i].src = response.data[startOfData].images[0];
+			
+			// set merchant of product and change it to Title case
+			document.getElementsByClassName("merchant")[i].innerHTML = "at " + toTitleCase(response.data[startOfData].listing.merchant);
+
+			// set deal title 
+			var dealTitle = document.getElementsByClassName("deal-item-title")[i];
+			dealTitle.innerHTML = response.data[startOfData].name;
+			clamp(dealTitle); // if deal title is too long, clamp it to show ellipsis
+			
+			// set how long ago deal was posted
+			// commented out for now, may be needed for activity feed in the future
+			//document.getElementsByClassName("deal-posted-date")[i].innerHTML = getTimeAgo(response.data[startOfData].lastCreated);
+			
+			// set current price with correct currency
+			document.getElementsByClassName("current-price")[i].innerHTML = getCurrency(response.data[startOfData]) + round(response.data[startOfData].listing.currentPrice, 2);
```

Comment:
Actually, we have jQuery so you do have a shorthand for getting class name.
i.e. `$('.current-price')` instead.
Not necessary to change, but just to take note so that you can write less code with the same outcome in the future 😄